docs/sdk/src/polkadot_sdk/xcm.rs

```diff
@@ -1,5 +1,69 @@
 //! # XCM
 //!
-//! @KiChjang @franciscoaguirre
-//! TODO: RFCs, xcm-spec, the future of the repo, minimal example perhaps, forward to where actual
-//! docs are hosted. <https://github.com/paritytech/polkadot-sdk-docs/issues/58>
+//! XCM, or Cross-Consensus Messaging, is a **language** to communicate **intentions** between
+//! **consensus systems**.
+//!
+//! ## Overview
+//!
+//! XCM is a standard, whose specification lives in the [xcm format repo](https://github.com/paritytech/xcm-format).
+//! It's agnostic both in programming language and blockchain platform, which means it could be used
+//! in Rust in Polkadot, or in Go or C++ in any other platform like Cosmos or Ethereum.
+//!
+//! It enables different consensus systems to communicate with each other in an expressive manner.
+//! Consensus systems include blockchains, smart contracts, and any other state machine that
+//! achieves consensus in some way.
+//!
+//! XCM is based on a virtual machine, the XCVM.
+//! An XCM program is a series of instructions, which get executed one after the other by the
```

Comment:
Need to also say that what we usually call an "XCM message" is actually an XCM program, otherwise the sudden introduction of "XCM program" here gets confusing.